src/Microsoft.ComponentDetection.Detectors/poetry/Contracts/PoetrySource.cs

```diff
@@ -14,6 +14,7 @@ public class PoetrySource
         public string reference { get; set; }
 
         [SuppressMessage("StyleCop.CSharp.NamingRules", "SA1300:ElementMustBeginWithUpperCaseLetter", Justification = "Deserialization contract. Casing cannot be overwritten.")]
+        [SuppressMessage("Naming", "CA1707:IdentifiersShouldNotContainUnderscores", Justification = "Deserialization contract. Casing cannot be overwritten.")]
         public string resolved_reference { get; set; }
```

Comment:
Use `DataMember` attributes[^1] denote the serialized name, and fix the name in our codebase.

[^1]: https://learn.microsoft.com/en-us/dotnet/api/system.runtime.serialization.datamemberattribute?view=net-6.0

Reply:
fixed. Also fixed suppression for SA1300 at several other places using the same.